### PR TITLE
Fixing memory leak in delta.c

### PR DESCRIPTION
**Pull request by castulo**
Fixing:
- Memory leak delta.c
- Dead code in thread_pool.c

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

### CODE REVIEW

File: src/lib/thread_pool.c [hunk 1]
```diff
@@ -91,12 +91,7 @@ struct tp *tp_start(int num_threads)
 	}
 
 	tp = malloc_or_die(sizeof(struct tp) + int_to_uint(num_threads) * sizeof(pthread_t *));
-
 	tp->num_threads = num_threads;
-	if (num_threads == 0) {
```
Comment (otaviobp) [inline src/lib/thread_pool.c:96]: There's a fix for that on #1564

File: src/swupd_lib/delta.c [hunk 2]
```diff
@@ -117,6 +117,7 @@ void apply_deltas(struct manifest *current_manifest)
 	DIR *dir = opendir(delta_dir);
 	if (!dir) {
 		/* No deltas available to apply. */
+		FREE(delta_dir);
```
Comment (otaviobp) [inline src/swupd_lib/delta.c:120]: +1 for this one

File: src/lib/thread_pool.c (hunk 1 above)
Comment (castulo) [reply src/lib/thread_pool.c:96]: I'll remove that from here then and will review #1564 😝

**Comment by castulo** [comment]
V: only fixing the memory leak

**Pull request merged**